pramen-py/tests/test_metastore/test_metastore.py

```diff
@@ -26,12 +26,108 @@
 from chispa.schema_comparer import SchemasNotEqualError
 from loguru import logger
 from pyhocon import ConfigFactory  # type: ignore
-from pyspark.sql import SparkSession
+from pyspark.sql import DataFrame, SparkSession
 
 from pramen_py import MetastoreReader, MetastoreWriter
 from pramen_py.models import InfoDateSettings, MetastoreTable, TableFormat
 
 
+def test_extract_dates_from_file_names(spark):
+    file_names = [
+        "pramen-py/tests_table/__HIDDEN_FILE__",
+        "pramen-py/tests_table/pramen_info_date=2022-04-20",
+        "pramen-py\\tests_table\\pramen_info_date=2022-04-21",
+        "pramen-py/tests_table\\pramen_info_date=2022-04-22",
+        "pramen-py/tests_table/pramen_info_date=2022-04-23",
+        "pramen-py/tests_table/test_metastore/pramen_date=2022-04-24",
```

Comment:
Nice tests! You can also add
```
"",
"2022-04-25",
"=2022-04-26="
"pramen_date=2022-04-27",
"info_date=2022-04-28",
"=2022-04-29",
"pramen_date=30-04-2022",
```
just to ensure incorrect folder names are ignored.